src/helpers.js

```diff
@@ -179,8 +179,34 @@ exports.allowedZero = function(val, char, caretPos, options) {
  * @param {val} string value to convert to a number
  * @param {options} Finput options object
  */
-exports.toNumber = function(val, options) {
-  return val && Number(val.replace(new RegExp(`[${options.thousands}]`, 'g'), ''));
+exports.toNumber = function(stringValue, options) {
+  if (!stringValue) return null;
+
+  // Number(...) accepts thousands ',' or '' and decimal '.' so we must:
+
+  // 1. Remove thousands delimiter to cover case it is not ','
+  // Cannot replace with ',' in case decimal uses this
+  stringValue = stringValue.replace(new RegExp(`[${options.thousands}]`, 'g'), '')
+  
+  // 2. Replace decimal with '.' to cover case it is not '.'
+  // Ok to replace as thousands delimiter removed above
+  stringValue = stringValue.replace(new RegExp(`[${options.decimal}]`, 'g'), '.')
+  return Number(stringValue);
```

Comment:
Something that just came to mind is that wrapping a string in the number constructor is going to cause rounding errors for values above `Number.MAX_SAFE_INTEGER`, we'll have to look at fixing this in a separate issue.
e.g. `Number("9007199254740991.45")` will result in `9007199254740991`